### PR TITLE
feat: replace interfaces for query and command with abstract records

### PR DESCRIPTION
**Pull request by rafalschmidt97**
Replace ICommand and IQuery with Command and Query (based on the approach from [CQRS.Mediatr.Lite](https://github.com/microsoft/CQRS.Mediatr.Lite/blob/a6f63cf62a5e2b1b48a55b7917ba036c8ae6f3b9/src/sdk/Command/Command.cs))

### CODE REVIEW

**Review by aszczepanski-allegro — APPROVED** [review]

**Review by darthkurak — APPROVED** [review]

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Allegro.Extensions.Cqrs.Abstractions.Queries;
+
+/// <summary>
+/// Cqrs base representation of query.
+/// </summary>
+public abstract record Query
```
Comment (rafalschmidt97) [inline src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs:8]: internal?

**Review by darthkurak — APPROVED** [review]

**Review by sowa705 — APPROVED** [review]

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    /// <summary>
+    /// Query identifier
+    /// </summary>
+    public string Id { get; } = Guid.NewGuid().ToString();
+}
+
+/// <summary>
+/// Cqrs base representation of query.
+/// </summary>
```
Comment (aszczepanski-allegro) [inline src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs:18]: nit: missing description of the `T` type param

Comment (aszczepanski-allegro) [reply src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs:18]: I guess it should be
```suggestion
<typeparam name="TResult">Type of data returned by query</typeparam>
```

**Review by aszczepanski-allegro — APPROVED** [review]

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs (hunk 2 above)
Comment (rafalschmidt97) [reply src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs:18]: fixed

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Commands/Command.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+/// <summary>
+/// Cqrs base representation of command.
+/// </summary>
+public abstract record Command
+{
+    /// <summary>
+    /// Command identifier
+    /// </summary>
+    public string Id { get; } = Guid.NewGuid().ToString();
```
Comment (ppiwow-apay) [inline src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Commands/Command.cs:13]: possbility to verride - virutal maybe?

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+public abstract record Query
+{
+    /// <summary>
+    /// Query identifier
+    /// </summary>
+    public string Id { get; } = Guid.NewGuid().ToString();
```
Comment (ppiwow-apay) [inline src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs:13]: same as in command virtual? possibility to pass it from external source (idempotency key?)

**Review by ppiwow-apay — APPROVED** [review]

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Commands/Command.cs (hunk 3 above)
Comment (rafalschmidt97) [reply src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Commands/Command.cs:13]: when someone will have a reason to add we will reconsider it

File: src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs (hunk 4 above)
Comment (rafalschmidt97) [reply src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs:13]: as above

Comment (rafalschmidt97) [reply src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs:13]: Imo they should create additional property for it to emphasise the behavior and reason

**Pull request merged**